### ISSUE TITLE
Contents of a script tag are ignored by the html parser

### ISSUE THREAD

**Issue by ramya-rao-a** [emmetio/html-matcher#2]
Contents of a script tag with type `text/html`, `text/template`, `text/x-template` are valid html.
The `parse` method ignores all script and style tags https://github.com/emmetio/html-matcher/blob/v0.3.2/index.js#L19

cc @sergeche

**Comment by ramya-rao-a** [comment]
@sergeche One way to fix it from the editor's side is to use `parse` with options where the `options.special` is just `['style']`

Wanted to check with you if there are any downsides in doing so.

Was the reason to exclude the style and script tags just an optimizing technique?

**Comment by sergeche** [comment]
A script tag can potentially contain code that could break HTML parsing:

```html
<div>
  <script>
  const tag = '<div ' + attrs + '></div>';
  // closing </div> can match with <div> before script.
  </script>
</div>
```

I think there’s should be an option for parser that should check if current tag is ”special”, e.g. check it via function, not an array of tag names

**Comment by sergeche** [comment]
Fixed in v1.0.0
